### PR TITLE
support metadata field in compute instance template

### LINKED ISSUE #859: Support metadata field in compute instance template
This is needed, for example, to set os login to true to instances to enable IAP access

### PR DESCRIPTION
**Pull request by xingao267**
Fixes #859

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -280,8 +280,6 @@ github.com/hashicorp/hcl/v2 v2.9.1 h1:eOy4gREY0/ZQHNItlfuEZqtcQbXIxzojlP301hDpna
 github.com/hashicorp/hcl/v2 v2.9.1/go.mod h1:FwWsfWEjyV/CMj8s/gqAuiviY72rJ1/oayI9WftqcKg=
 github.com/hashicorp/memberlist v0.1.0/go.mod h1:ncdBp14cuox2iFOq3kDiquKU6fqsTBc3W6JvZwjxxsE=
 github.com/hashicorp/serf v0.0.0-20160124182025-e4ec8cc423bb/go.mod h1:h/Ru6tmZazX7WO/GDmwdpS975F019L4t5ng5IgwbNrE=
```
Comment (MartinPetkov) [inline go.sum:282]: Nit: If you're not updating Go files, this should be a separate PR.

File: templates/tfengine/components/resources/compute_instance_templates/main.tf [hunk 2]
```diff
@@ -71,6 +71,16 @@ EOF
     {{end -}}
   }
   {{end -}}
+
+
+  {{if has . "metadata" -}}
```
Comment (MartinPetkov) [inline templates/tfengine/components/resources/compute_instance_templates/main.tf:76]: Could this be replaced with `hclField`?

Comment (xingao267) [reply templates/tfengine/components/resources/compute_instance_templates/main.tf:76]: Map is currently not supported by `hclField`. https://github.com/GoogleCloudPlatform/healthcare-data-protection-suite/issues/666

File: go.sum (hunk 1 above)
Comment (xingao267) [reply go.sum:282]: https://github.com/GoogleCloudPlatform/healthcare-data-protection-suite/pull/861

File: templates/tfengine/components/resources/compute_instance_templates/main.tf (hunk 2 above)
Comment (MartinPetkov) [reply templates/tfengine/components/resources/compute_instance_templates/main.tf:76]: I just tried and you can still use `hcl`:
```
{{if has . "metadata" -}}
   metadata = {
      {{hcl .metadata}}
   }
{{end -}}
```

Comment (xingao267) [reply templates/tfengine/components/resources/compute_instance_templates/main.tf:76]: Done

**Review by MartinPetkov — APPROVED** [review]
